### PR TITLE
Rubocop touch ups and GHA

### PR DESCRIPTION
**Pull request by astley92**
### Pull Request Checklist

* [x] Pull request is based on the `main` branch
* [x] Pull request includes a [sign off](../CONTRIBUTING.md#sign-off)

Signed-off-by: Blake Astley <astley92@hotmail.com>

### What

This is a VERY large PR that basically does nothing but formatting 😅 
I'm not sure how we feel about PR's that change this much and happy to split this up into more digestible PR's if anyone would prefer.
Basically this just cleans up our `.rubocop.yml` file, re-enables/configures a bunch of cops and gets everything passing.
`rubocop -A` was only used for very safe cop violations, all others were done manually.
The individual cop configuration was totally just my preference, totally happy to change if anyone has other thoughts.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -7,48 +7,58 @@ AllCops:
...
-    - app/controllers/users_controller.rb
-    - app/controllers/events_controller.rb
-    - app/controllers/registrations_controller.rb
-    - app/controllers/graphql_controller.rb
-    - app/models/header.rb
-    - app/models/data_set.rb
-    - app/models/team_entry_report.rb
-    - app/services/demographics_report.rb
-    - test/models/competition_test.rb
+    - vendor/bundle/**/*
```
Comment (astley92) [inline .rubocop.yml:13]: Apparently when using the `setup-ruby@v1` github action, rubocop will try pull in config files from the vendor files. Excluding them to prevent them from being picked up

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -3,6 +3,17 @@ name: CI
 on: pull_request
 
 jobs:
+  rubocop:
+    name: rubocop
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (bmpickford) [inline .github/workflows/ci.yml:10]: nitpick:
```suggestion
      - uses: actions/checkout@v3
```

**Review by bmpickford — APPROVED** [review]
didn't check all files, just spot checking but looks good as far as I can tell

**Review by 00Chaotic — APPROVED** [review]
Approving, but considering it's such a (simple but still) massive change, perhaps we should hold off on merging until we can test releases. I can see a few different PRs being put up so if we merge multiple changes before we have the relevant access/means to test them, it'll be a headache to troubleshoot if anything goes wrong.

**Comment by bmpickford** [comment]
> Approving, but considering it's such a (simple but still) massive change, perhaps we should hold off on merging into main until we can test releases. I can see a few different PRs being put up so if we merge multiple changes before we have the relevant access/means to test them, it'll be a headache to troubleshoot if anything goes wrong.

I think at this stage building momentum is more important and we shouldn't be introducing blockers. None of the PR's are particularly risky imo or at least risky enough to block merging into main.

Hopefully we can get a test env setup soon but we should be testing locally at least in the meantime to make sure nothing is significantly broken. And maybe to mitigate any risk on our first release we can do a bit of mob testing but I think if we block merges the PR's are just going to become stale and we lose what little momentum we might have.
